src/ExtranetBundle/Service/ExtranetUserProvider.php

```diff
@@ -37,53 +37,43 @@ public function __construct(RequestStack $requestStack)
      */
     public function getActiveUser()
     {
-        if (false === $this->hasSession() || false === $this->sessionIsStarted()) {
-            return null;
+        $session = $this->getStartedSession();
+
+        if (null === $session) {
+            return \TdbDataExtranetUser::GetNewInstance();
         }
 
-        $user = $this->getSession()->get(self::SESSION_KEY_NAME);
+        $user = $session->get(self::SESSION_KEY_NAME);
         if (null === $user) {
             $this->reset();
         }
 
-        return $this->getSession()->get(self::SESSION_KEY_NAME);
+        return $session->get(self::SESSION_KEY_NAME);
     }
 
     /**
      * {@inheritdoc}
      */
     public function reset()
     {
-        $this->getSession()->set(self::SESSION_KEY_NAME, \TdbDataExtranetUser::GetNewInstance());
-    }
+        $session = $this->getStartedSession();
+        if (null === $session) {
+            return;
+        }
 
-    /**
-     * @return SessionInterface|null
-     */
-    private function getSession()
-    {
-        return $this->requestStack->getCurrentRequest()->getSession();
+        $session->set(self::SESSION_KEY_NAME, \TdbDataExtranetUser::GetNewInstance());
     }
 
-    /**
-     * @return bool
-     */
-    private function hasSession()
+    private function getStartedSession(): ?SessionInterface
     {
-        $hasSession = false;
         $request = $this->requestStack->getCurrentRequest();
-        if (null !== $request) {
-            $hasSession = $request->hasSession();
+        if (null === $request) {
+            return null;
+        }
+        if (false === $request->getSession()->isStarted()) {
+            return null;
         }
```

Comment:
We should check for `$request->hasSession()` before getting the session for the request as it might not be set.

Reply:
ok